website/blog/2018-02-23-1.11.0.md

```diff
@@ -0,0 +1,1006 @@
+```scss
+// Before
+$theme: blue;
+
+a {
+  @mixin $(theme) -colors;
+}
+
+// After
+$theme: blue;
+
+a {
+  @mixin $(theme)-colors;
+}
+```
+
+#### Support YAML front-matter block in SCSS ([#3802] by [@evilebottnawi])
+
+Jekyll relies on YAML front-matter to detect SCSS files, but Prettier was removing it, which would break Jekyll sites when deployed. The YAML front-matter is now preserved, which preserves styling:
+
+<!-- prettier-ignore -->
+```scss
+---
+title: Title
+description: Description
+---
+
+a {
+  color: red;
+}
+```
+
+### TypeScript
+
+#### Support definite assignment (`!:`) ([#4020] by [@ikatyang])
+
+TypeScript 2.7 added a new feature called definite assignment that can be used in class property initializers. It informs TypeScript that even though a class property does not have an initializer, we can be sure that at runtime it will have a value, rather than `undefined`:
+
+<!-- prettier-ignore -->
+```ts
+class MyComponent {
+  ngModel!: ng.INgModelController;
+}
+```
+
+Prettier 1.10.2 couldn't parse this syntax:
+
+<!-- prettier-ignore -->
+```ts
+SyntaxError: '=' expected. (2:10)
+  1 | class MyComponent {
+> 2 |   ngModel!: ng.INgModelController;
+    |          ^
+  3 | }
+```
+
+But it is now supported in Prettier 1.11.0.
+
+#### Support `unique symbol` ([#3967] by [@ikatyang])
+
+TypeScript 2.7 also added a new `unique symbol` type:
+
+<!-- prettier-ignore -->
+```ts
+interface SymbolConstructor {
+  readonly iterator: unique symbol; 
+}
+```
+
+Prettier 1.10.2 couldn't parse this syntax:
+
+<!-- prettier-ignore -->
+```ts
+SyntaxError: ';' expected. (2:29)
+  1 | interface SymbolConstructor {
+> 2 |   readonly iterator: unique symbol; 
+    |                             ^
+  3 | }
+  4 | 
+```
+
+But it is now supported in Prettier 1.11.0.
+
+#### Support condition types ([#4006] by [@Cryrivers])
+
+TypeScript 2.8 will add support for conditional types, which look like ternaries but appear in a type position:
+
+<!-- prettier-ignore -->
+```ts
+export type DeepReadonly<T> =
+    T extends any[] ? DeepReadonlyArray<T[number]> :
+    T extends object ? DeepReadonlyObject<T> :
+    T;
+```
+
+Prettier 1.10.2 couldn't parse these:
+
+<!-- prettier-ignore -->
+```ts
+SyntaxError: ';' expected. (2:7)
+  1 | export type DeepReadonly<T> =
+> 2 |     T extends any[] ? DeepReadonlyArray<T[number]> :
+    |       ^
+  3 |     T extends object ? DeepReadonlyObject<T> :
+  4 |     T;
+```
+
+But this syntax is supported in Prettier 1.11.0.
+
+### JavaScript
+
+#### Ternaries were indented incorrectly with `--use-tabs` ([#3745] by [@ikatyang])
+
+Because tabs can be represented as different widths, prettier was sometimes indenting ternaries incorrectly when using tabs. Now we calculate the indentation properly:
+
+<!-- prettier-ignore -->
+```js
+// Before:
+const abc = props.something
+	? xyz
+	: {
+			prop1: a,
+			prop2: b,
+			prop3: false,
+		};
+
+// After:
+const abc = props.something
+	? xyz
+	: {
+			prop1: a,
```

Comment:
Why 3 tabs here?

Reply:
So that it's aligned when tabs are rendered as two spaces, I believe.

Reply:
There's a lot of discussion about it here: https://github.com/prettier/prettier/pull/3745